File: Source/Aladdin.ASN1.ISO/ASN1/ISO/PKCS/ContentInfo.cs
using System;

//	ContentInfo ::= SEQUENCE {
//		contentType				 OBJECT IDENTIFIER,
//		content		[0] EXPLICIT ANY DEFINED BY contentType 
//	}

namespace Aladdin.ASN1.ISO.PKCS
{
	public class ContentInfo : Sequence
	{
		// информация о структуре
		private static readonly ObjectInfo[] info = new ObjectInfo[] { 

			new ObjectInfo(new ObjectCreator<ObjectIdentifier>().Factory(), Cast.N,	Tag.Any			), 
			new ObjectInfo(    ImplicitCreator				    .Factory  , Cast.E,	Tag.Context(0)	), 
		}; 
		// конструктор при раскодировании
		public ContentInfo(IEncodable encodable) : base(encodable, info) {}

		// конструктор при закодировании
		public ContentInfo(ObjectIdentifier contentType, IEncodable content) : 
			base(info, contentType, content) {}

		public ObjectIdentifier	ContentType	{ get { return (ObjectIdentifier)this[0]; } } 
		public IEncodable		Inner		{ get { return					 this[1]; } }
	}
}

File: Source/Aladdin.ASN1.ISO.Java/src/aladdin/asn1/iso/pkcs/ContentInfo.java
package aladdin.asn1.iso.pkcs;
import aladdin.asn1.*; 
import java.io.*;

//	ContentInfo ::= SEQUENCE {
//		contentType				 OBJECT IDENTIFIER,
//		content		[0] EXPLICIT ANY DEFINED BY contentType 
//	}

public final class ContentInfo extends Sequence<IEncodable>
{
    private static final long serialVersionUID = -6967099234116925051L;
    
	// информация о структуре
	private static final ObjectInfo[] info = new ObjectInfo[] { 

		new ObjectInfo(new ObjectCreator(ObjectIdentifier.class).factory(), Cast.N,	Tag.ANY			), 
		new ObjectInfo(    ImplicitCreator						.factory  , Cast.E,	Tag.context(0)	), 
	}; 
	// конструктор при раскодировании
	public ContentInfo(IEncodable encodable) throws IOException { super(encodable, info); }

	// конструктор при закодировании
	public ContentInfo(ObjectIdentifier contentType, IEncodable content) 
	{
		super(info, contentType, content); 
	}
	public final ObjectIdentifier contentType() { return (ObjectIdentifier)get(0); } 
	public final IEncodable		  inner	     () { return				   get(1); }
}
